generator/borders.cpp

```diff
@@ -232,43 +230,65 @@ bool LoadCountriesList(string const & baseDir, CountriesContainer & countries)
   return !countries.IsEmpty();
 }
 
-void GeneratePackedBorders(string const & baseDir)
+void GeneratePackedBorders(std::string const & baseDir)
 {
   PackedBordersGenerator generator(baseDir);
   ForEachCountry(baseDir, generator);
   generator.WritePolygonsInfo();
 }
 
-void UnpackBorders(string const & baseDir, string const & targetDir)
+void DumpBorderToPolyFile(std::string const & filePath, std::string const & mwmName,
+                          std::vector<std::vector<m2::PointD>> const & polygons)
+{
+  CHECK(!polygons.empty(), ());
+
+  std::ofstream poly(filePath);
+  CHECK(poly.good(), ());
+
+  poly << mwmName << std::endl;
+  size_t polygonId = 1;
+  for (auto const & points : polygons)
+  {
+    poly << polygonId << std::endl;
+    ++polygonId;
+    for (auto const & point : points)
+    {
+      ms::LatLon const ll = MercatorBounds::ToLatLon(point);
+      poly << "    " << std::scientific << ll.m_lon << "    " << ll.m_lat << std::endl;
+    }
+    poly << "END" << std::endl;
+  }
+
+  poly << "END" << std::endl;
+  poly.close();
+}
+
+void UnpackBorders(std::string const & baseDir, std::string const & targetDir)
 {
   if (!Platform::IsFileExistsByFullPath(targetDir) && !Platform::MkDirChecked(targetDir))
     MYTHROW(FileSystemException, ("Unable to find or create directory", targetDir));
 
-  vector<storage::CountryDef> countries;
+  std::vector<storage::CountryDef> countries;
   FilesContainerR reader(base::JoinPath(baseDir, PACKED_POLYGONS_FILE));
   ReaderSource<ModelReaderPtr> src(reader.GetReader(PACKED_POLYGONS_INFO_TAG));
   rw::Read(src, countries);
 
   for (size_t id = 0; id < countries.size(); id++)
   {
-    ofstream poly(base::JoinPath(targetDir, countries[id].m_countryId + ".poly"));
-    poly << countries[id].m_countryId << endl;
+    std::string const filePath = base::JoinPath(targetDir, countries[id].m_countryId + ".poly");
```

Comment:
Can we instead pass targetDir and mwmName?
From what I see it's very confusing (and probably wrong wrt the generation process) to have a .poly file named differently from the mwmName written there.